/*
 *  Copyright (c) 2024 Qualcomm Innovation Center, Inc. All rights reserved.
 *  SPDX-License-Identifier: BSD-3-Clause-Clear
 */

syntax = "proto3";

package satcomStub;
import "google/protobuf/empty.proto";
import "common_simulation.proto";

/*******************************************************************
* Services Definition                                             *
*******************************************************************/

/*
* This service is used to execute NtnManager related APIs.
*/

service NtnManager {
    rpc InitService(google.protobuf.Empty) returns (GetServiceStatusReply) {}
}

/*******************************************************************
 * Messages Definition                                             *
 *******************************************************************/

enum ServiceStatus {
    SERVICE_UNAVAILABLE = 0;
    SERVICE_AVAILABLE = 1;
    SERVICE_FAILED = 2;
}

message GetServiceStatusReply {
    ServiceStatus service_status = 1;
    int32 delay = 2;
}
/*******************************************************************
 * Messages Definition                                             *
 *******************************************************************/
